src/file_request_handler.cc

```diff
@@ -92,12 +92,12 @@ void FileRequestHandler::getInfo(const char* filename, UpnpFileInfo* info)
     std::string rh = getValueOrDefault(params, RESOURCE_HANDLER);
 
     // determining which resource to serve
-    size_t res_id = 0;
     auto res_id_it = params.find(URL_RESOURCE_ID);
-    if (res_id_it != params.end() && res_id_it->second != URL_VALUE_TRANSCODE_NO_RES_ID)
-        res_id = std::stoi(res_id_it->second);
-    else
-        res_id = SIZE_MAX;
+    auto res_id = size_t { [&]() {
+        if (res_id_it != params.end() && res_id_it->second != URL_VALUE_TRANSCODE_NO_RES_ID)
+            return size_t(std::stoi(res_id_it->second));
+        return std::numeric_limits<size_t>::max();
+    }() };
```

Comment:
I prefer initializing with max value and the following if. The nesting is quite unreadable to me.